packages/scheduler/src/forks/SchedulerHostConfig.default.js

```diff
@@ -16,10 +16,24 @@ export let requestPaint;
 export let getCurrentTime;
 export let forceFrameRate;
 
+const isWindowUndefined = typeof window === 'undefined';
+
+const performance = isWindowUndefined ? global.performance : window.performance;
```

Comment:
Why is this needed? Why not just use `performance` as a global in the first place?

Reply:
This is based on a comment in the previous code.

```
// Capture local references to native APIs, in case a polyfill overrides them.
```

Reply:
I wonder if all environments that don't have `window` are guaranteed to have `global`, or not. We don't already use `global` in this file yet.

Reply:
For example what if this code is loaded in a worker. Would it crash?

Reply:
I would feel better about refactoring to read globals directly from the global scope (without a qualifier) and aliasing them. Kind of like it was before https://github.com/facebook/react/commit/3f2cafe8bef0922525b445f7bd2e23ea75c817cf.

Reply:
Good thinking. I refactored the code. I had to refactor the browser test too, `window` is not considered as global under test environment.